app/models/immunization.rb

```diff
@@ -0,0 +1,79 @@
+class Immunization
+  class << self
+    @tradenames = nil
+
+    def tradenames
+      @tradenames ||= load_tradenames
+    end
+
+    def load_tradenames
+      f = File.open(Rails.root.join('app', 'assets', 'iisstandards_tradename.xml'))
+      @tradenames = Nokogiri::XML(f)
+    end
+  end
+
+  def initialize(fhir_immunization)
+    @fhir_immunization = fhir_immunization
+    @occurrence = occurrence
```

Comment:
Everything should just be delegated to the FHIR model, keeping copies pulled out of the FHIR data can lead to inconsistency (we ran into this a lot when designing the HC library)

Reply:
We don't have a data store here so unlikely to lead to inconsistency with data. The UI model is much easier to work with in the front end than fhir bundles, which are exchange formats and not display friendly. Esp if multiple frontends ie mobile friendly version, etc. If we really want to move this to frontend, would put in a separate task to redo this.

Reply:
I'm not sure I understand the purpose of these attributes since it doesn't look like things like any of them (e.g. `@occurrence`) are ever used and I don't see any attribute readers defined for them.

I can understand the helper methods, below for pulling out the values from the FHIR model object, but these copies seem irrelevant.

Reply:
The assignment here seems to be backwards. I think you can just get rid of the instance variables and use the methods. If you really want to cache the values the methods should conditionally set/return the instance variables.